src/Multipart/AbstractUploadManager.php

```diff
@@ -49,16 +48,21 @@
     /** @var UploadState State used to manage the upload. */
     protected $state;
 
+    /** @var string Exception type thrown for the upload */
+    protected $exception;
+
     /**
      * @param Client $client
+     * @param string $exception
      * @param array  $config
      */
-    public function __construct(Client $client, array $config = [])
+    public function __construct(Client $client, array $config = [], $exception = null)
```

Comment:
Yeah, I also feel weird about leaving the scalar argument there. I hesitated to put it in `config` array originally because all options for `config` is documented [publicly](http://docs.aws.amazon.com/aws-sdk-php/v3/guide/service/s3-multipart-upload.html#configuration), and I'm not that sure whether `exception_class` parameter is appropriate (or make sense?) to document there.

I moved it to the `config` class with renaming in the latest commit :)